tests/unit/client_test.py

```
@@ -249,3 +252,289 @@ def test_set_default_max_query_payment_non_finite_value(invalid_amount):
+
+        mock_set_nodes.assert_called_once()
+        assert returned is client
+
+    client.close()
+
+def test_warning_when_grpc_deadline_exceeds_request_timeout():
+    """Warn when grpc_deadline is greater than request_timeout."""
+    client = Client.for_testnet()
+    client.set_request_timeout(2)
+
+    with pytest.warns(UserWarning):
```

Ruff (0.14.14)

[warning] `pytest.warns(UserWarning)` is too broad, set the `match` parameter or use a more specific warning

(PT030)